bigquery_etl/docs/derived_datasets/generate_derived_dataset_docs.py

```diff
@@ -57,6 +58,11 @@ def generate_derived_dataset_docs(out_dir, project_dir):
                             print(error)
                 if VIEW_FILE in files:
                     source_urls["View Definition"] = f"{SOURCE_URL}/{root}/{VIEW_FILE}"
+                    view_file = os.path.join(root, VIEW_FILE)
+
```

Comment:
Just to consider the option where parsing SQL is not required, you could do something like this:

```python
import tempfile
...
if VIEW_FILE in files:
    source_urls["View Definition"] = f"{SOURCE_URL}/{root}/{VIEW_FILE}"

    # find the dataset/table information
    path = Path(root)
    dataset = path.parent.parent.name
    view = path.parent.name

    # create a temp file to pass into dryrun
    temp_sqlfile = Path(tempfile.mkdtemp()) / "view.sql"
    temp_sqlfile.write_text(f"select * from {dataset}.{view}")
    referenced_tables = DryRun(sqlfile=tmp_sqlfile)
```

Reply:
I think we need to pass in the original view file so that it can be tested against the SKIP list. But I'll add this logic in the DryRun class with `strip_dml` flag to get the referenced tables

Reply:
After implementing this method, I found that it returns a lot of unfound table errors, like: `'message': 'Not found: Dataset moz-fx-data-shared-prod:telemetry.clients_scalar_aggregates_v1 was not found in location US', 'domain': 'global', 'reason': 'notFound'`
After comparing the names of sql directories in bigquery-etl to BigQuery datasets, I found that there's not a direct 1:1 correlation between the dir names, so extracting view names from directories like this would not work